app/models/tag.rb

```diff
@@ -20,15 +20,18 @@ class Tag < ActsAsTaggableOn::Tag
             format: /\A#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})\z/, allow_nil: true
   validates :category, inclusion: { in: ALLOWED_CATEGORIES }
 
-  validate :validate_alias
-  validate :validate_name
+  validate :validate_alias_for, if: proc { alias_for.present? }
+  validate :validate_name, if: proc { name.present? }
```

Comment:
This was shadowing the `ActsAsTaggableOn::Tag` default validator, so that if one tries to do `Tag.create!` it'll end up in a bug:

```ruby
[1] pry(main)> Tag.create!
NoMethodError: undefined method `length' for nil:NilClass
```

now it correctly displays the validation error:

```ruby
[1] pry(main)> Tag.create!
ActiveRecord::RecordInvalid: Validation failed: Name can't be blank
```